std/os/time.zig

```diff
@@ -31,8 +31,8 @@ pub fn sleep(nanoseconds: u64) void {
 
 pub fn posixSleep(seconds: u63, nanoseconds: u63) void {
     var req = posix.timespec{
-        .tv_sec = seconds,
-        .tv_nsec = nanoseconds,
+        .tv_sec = @intCast(isize, seconds),
```

Comment:
This introduces a change to the API of this function: previously any value that fit inside of a u63 was allowed. After this change, the integer range between pointer size and u63 is an assertion failure.

I think it's good that the sleep function cannot fail. I also think it should have legal behavior for the entire input range. Here is my proposal to accomplish this:
 * Add documentation comment to note that posixSleep() may have spurious wakeups. This is already the case, as noted by the EINVAL case.
 * If the number of seconds/nanoseconds exceeds the capacity of the integers in the `timespec` struct, do the sleep syscall for the maximum duration, and then simply return, treating it as a spurious wakeup.

Reply:
I've implemented another solution, we just slice and dice the number of seconds and do N calls to `nanosleep` until we've slept for the desired amount of time.

Reply:
Ah I'm sorry I didn't see this comment until later, and I found the new code harder to reason about and verify than my idea and so I modified the logic when merging (I also added doc comments and fixed the other sleep function). If you feel strongly about it and are willing to put up another PR I'll merge it.

Reply:
> If you feel strongly about it and are willing to put up another PR I'll merge it.

Nevermind, sleeping for such a long time is quite a niche use-case that any behavior (as long as it's documented) is fine.